docs/cpp/lldb-mi.md

```diff
@@ -0,0 +1,77 @@
+---
+Order: 15
+Area: cpp
+TOCTitle: Setup Help for LLDB-MI
+ContentId: 1e84d196-397f-4dbb-9746-06f15766d83e
+PageTitle: How to setup debugging on macOS with LLDB-MI
+DateApproved: 2/01/2020
+MetaDescription: How to setup debugging on macOS with LLDB-MI
+---
+# Setup
+The debug adapter for the C/C++ extension utilizes the machine interface mode for both gdb and lldb. To use this interface in lldb, we utilize `lldb-mi`. We build and ship this executable which has a dependency on `LLDB.framework`.
+
+## Prerequisites
+The `lldb-mi` executable requires `LLDB.framework` to run.
+
+### How to obtain `LLDB.framework`
+You can get `LLDB.framework` from one of the two choices below:
+- `XCode`
+    1. Open the `Apple App Store`
+    2. Search for `XCode`
+    3. Install.
+
+- `XCode Command Line Tools`
+    1. Open a terminal
+    2. Run `xcode-select --install`
+    3. Confirm prompt
+
+## Example default `launch.json`
+```
+{
+    "name": "Launch (lldb)"
+    "type": "cppdbg",
+    "request": "launch",
+    "program": "${workspaceFolder}/a.out"
+    "args": [],
+    "stopAtEntry": false,
+    "cwd": "${workspaceFolder}
+    "environment": [],
+    "externalConsole": false
+}
+```
+
+# Additional Information
+
+## Developer Tool Access needs to take control of another process for debugging to continue.
+
+![Developer Tool Access problem](images/debugger/DeveloperToolsAccess.png)
+
+If you get this prompt, will have you type in your password to allow debugging.
+
+If you wish to permanently dismiss this prompt, you can run the following command in a terminal:
+
+```
+sudo /usr/sbin/DevToolsSecurity --enable
+```
+
+## Non-default LLDB.framework paths or custom lldb-mi
+If you wish to use a LLDB.framework that is not installed the way described above, you can copy the `lldb-mi` executable in `~/.vscode/extensions/ms.vscode-cpptools-<version>/debugAdapters/lldb/bin` right next to where the `LLDB.framework` is.
+
+Then add the full path to `lldb-mi` to `miDebuggerPath` in your `launch.json` configuration.
+
+## Example
+Assuming you have `LLDB.framework` at `"/Users/default/example/"`
```

Comment:
Instead of `Assuming` make this a `You have LLDB.Framework located at ...` and then have a step to copy `lldb-mi` in there also.